Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -1092,10 +1092,22 @@ bool ctkDICOMBrowser::confirmDeleteSelectedUIDs(QStringList uids)
   ctkMessageBox confirmDeleteDialog;
   QString message = tr("Do you want to delete the following selected items?");
 
+  // calculate maximum number of rows that fit in the browser widget to have a reasonable limit
+  // on the items to show in the dialog
+  int browserHeight = this->geometry().height();
+  int patientsTableRowHeight = d->dicomTableManager->patientsTable()->tableView()->rowHeight(0);
+  int maxNumberOfPatientsToShow = browserHeight / patientsTableRowHeight - 1; // Subtract 1 due to the do not show checkbox
+
   // add the information about the selected UIDs
   int numUIDs = uids.size();
   for (int i = 0; i < numUIDs; ++i)
   {
+    if (i >= maxNumberOfPatientsToShow)
+    {
+      message += QString("\n(and %1 more)").arg(numUIDs - maxNumberOfPatientsToShow);
```

Comment:
Use `tr()` and explain the message using a translator's comment (it shows up in Qt translator and Weblate) - indicated by `//:`

```suggestion
      message += QString("\n") + tr("(and %1 more)").arg(numUIDs - maxNumberOfPatientsToShow);  //: displayed when there are additional DICOM items to delete that do not fit on screen
```

Reply:
Thanks! Yes I was wondering about the translation, if it worked or not in CTK. OK I'll do this.